request-interfaces.go

```diff
@@ -104,6 +108,16 @@ type RealPathFileLister interface {
 	RealPath(string) (string, error)
 }
 
+// ReadlinkFileLister is a FileLister that implements the Readlink method.
+// By implementing the Readlink method, it is possible to give a better
+// response than via the default FileLister, since Readlink can return
+// a better result than FileLister (which is limited to os.FileInfo, whose
+// Name() should only return the base name of a file)
```

Comment:
I leave this comment block as an exercise for the reader to improve with sembr. I think it’s probably fine and explicitly spelling out what is important.